parser-typechecker/src/Unison/Hashing/V2/ABT.hs

```diff
@@ -78,27 +79,25 @@ hash :: forall f v a h . (Functor f, Hashable1 f, Eq v, Show v, Ord h, Accumulat
      => Term f v a -> h
 hash = hash' [] where
   hash' :: [Either [v] v] -> Term f v a -> h
-  hash' env (Term _ _ t) = case t of
-    Var v -> maybe die hashInt ind
+  hash' env = \case
+    Var' v -> maybe die hashInt ind
       where lookup (Left cycle) = v `elem` cycle
             lookup (Right v') = v == v'
             ind = findIndex lookup env
             hashInt :: Int -> h
             hashInt i = Hashable.accumulate [Hashable.Nat $ fromIntegral i]
             die = error $ "unknown var in environment: " ++ show v
                         ++ " environment = " ++ show env
-    Cycle (AbsN' vs t) -> hash' (Left vs : env) t
-    -- Cycle t -> hash' env t
-    Abs v t -> hash' (Right v : env) t
-    Tm t -> Hashable.hash1 (hashCycle env) (hash' env) t
+    Cycle' vs t -> Hashable.hash1 (hashCycle vs env) undefined t
```

Comment:
I think the need to use `undefined` here means we should probably re-visit this typeclass implementation at some point in the future, at a glance I don't understand why `hash1` needs two separate hashing function args, but maybe there's a reason.

Reply:
Yeah, it's a bit of a weird type class for sure.